config/decoder.go

```diff
@@ -231,6 +224,23 @@ func (d *decoder) mapping(childKey string, value reflect.Value, def string) erro
 	return nil
 }
 
+// Sets value to an interface type.
+func (d *decoder) iface(key string, value reflect.Value, def string) error {
+	v := d.getGlobalProvider().Get(key)
+
+	if !v.HasValue() || v.Value() == nil {
+		return nil
+	}
+
+	src := reflect.ValueOf(v.Value())
+	if src.Type().Implements(value.Type()) {
+		value.Set(src)
+		return nil
+	}
+
```

Comment:
It is the msft style and I think google java style: line breaks after closing non terminal brackets :)